client/session.go

```diff
@@ -153,18 +153,27 @@ func (s *session) Open() error {
 	s.seriesIteratorsPool.Init()
 	s.Unlock()
 
-	currTopologyMap := s.topo.GetAndSubscribe(s.topoMapCh)
+	currTopologyMap, topologyWatch, err := s.topo.GetAndSubscribe()
+	if err != nil {
+		return err
```

Comment:
I see, maybe I can do the s.topo.GetAndSubscribe() and s.setTopologyMap(currTopologyMap) in the beginning of Open()
before the line s.state = stateOpen?